example/workflows.yml

```diff
@@ -535,64 +535,98 @@ IngestGranule:
-        Next: TranslateMessage
-      TranslateMessage:
-        Type: Task
-        Resource: ${CNMToCMALambdaFunction.Arn}
-        Retry:
-          - ErrorEquals:
-              - States.ALL
-            IntervalSeconds: 10
-            MaxAttempts: 2
-        Catch:
-          - ErrorEquals:
+  Comment: 'Tests Workflow from Kinesis Stream'
+  StartAt: TranslateMessage
+  States:
+    TranslateMessage:
+      CumulusConfig:
+        cumulus_message:
+          outputs:
+            - source: '{$.cnm}'
+              destination: '{$.meta.cnm}'
+            - source: '{$}'
+              destination: '{$.payload}'
+      Type: Task
+      Resource: ${CNMToCMALambdaFunction.Arn}
+      Retry:
+        - ErrorEquals:
             - States.ALL
-            ResultPath: '$.exception'
-            Next: StopStatus
-        Next: SyncGranule
-      SyncGranule:
-        CumulusConfig:
-          provider: '{$.meta.provider}'
-          buckets: '{$.meta.buckets}'
-          collection: '{$.meta.collection}'
-          downloadBucket: '{$.meta.buckets.private.name}'
-          stack: '{$.meta.stack}'
-          cumulus_message:
-            outputs:
-              - source: '{$.granules}'
-                destination: '{$.meta.input_granules}'
-              - source: '{$}'
-                destination: '{$.payload}'
-        Type: Task
-        Resource: ${SyncGranuleLambdaFunction.Arn}
-        Retry:
-          - ErrorEquals:
-              - States.ALL
-            IntervalSeconds: 10
-            MaxAttempts: 3
-        Catch:
-          - ErrorEquals:
+          IntervalSeconds: 10
+          MaxAttempts: 2
+      Catch:
+        - ErrorEquals:
+          - States.ALL
+          ResultPath: '$.exception'
+          Next: CnmResponse
+      Next: Report
+    Report:
+      CumulusConfig:
+        cumulus_message:
+          input: '{$}'
+      ResultPath: null
+      Type: Task
+      Resource: ${sf2snsStartLambdaFunction.Arn}
+      Next: SyncGranule
+    SyncGranule:
+      CumulusConfig:
+        provider: '{$.meta.provider}'
+        buckets: '{$.meta.buckets}'
+        collection: '{$.meta.collection}'
+        downloadBucket: '{$.meta.buckets.private.name}'
+        stack: '{$.meta.stack}'
+        cumulus_message:
+          outputs:
+            - source: '{$.granules}'
+              destination: '{$.meta.input_granules}'
+            - source: '{$}'
+              destination: '{$.payload}'
+      Type: Task
+      Resource: ${SyncGranuleLambdaFunction.Arn}
+      Retry:
+        - ErrorEquals:
             - States.ALL
-            ResultPath: '$.exception'
-            Next: StopStatus
-        Next: StopStatus
-      StopStatus:
-        Type: Task
-        Resource: ${sf2snsEndLambdaFunction.Arn}
-        Catch:
-          - ErrorEquals:
+          IntervalSeconds: 10
+          MaxAttempts: 3
+      Catch:
+        - ErrorEquals:
+          - States.ALL
+          ResultPath: '$.exception'
+          Next: StopStatus
```

Comment:
I guess the PR for Failure case will handle workflow errors and report back to CnmResponse stream?

Reply:
Correct, in an upcoming PR